arches/app/models/migrations/11848_alter_spatialview_geometrynode.py

```diff
@@ -0,0 +1,156 @@
+                        execute sv_perform;
+                    end if;
+
+                    return old;
+
+                elsif tg_op = 'INSERT' then
+                    if new.isactive = true then
+                        sv_perform := sv_perform || format(
+                            'select __arches_create_spatial_view(%L, %L::uuid, %L::jsonb, %L, %L, %L, %L);'
+                            , new.slug
+                            , new.geometrynodeid
+                            , new.attributenodes
+                            , new.schema
+                            , new.description
+                            , new.ismixedgeometrytypes
+                            , new.languageid);
+                    end if;
+
+                    if sv_perform <> '' then
+                        execute sv_perform;
+                    end if;
+
+                    return new;
+
+                elsif tg_op = 'UPDATE' then
+
+                    if new.isactive = true then
+                        sv_perform := sv_perform || format(
+                            'select __arches_update_spatial_view(%L, %L, %L, %L::uuid, %L::jsonb, %L, %L, %L);'
+                            , old.slug
+                            , old.schema
+                            , new.slug
+                            , new.geometrynodeid
+                            , new.attributenodes
+                            , new.schema
+                            , new.description
+                            , new.ismixedgeometrytypes
+                            , new.languageid);
+                    else
+                        sv_perform := sv_perform || format(
+                            'select __arches_delete_spatial_view(%L,%L);'
+                            , old.slug
+                            , old.schema);
+                    end if;
+
+                    if sv_perform <> '' then
+                        execute sv_perform;
+                    end if;
+
+                    return new;
+                end if;
+            end;
+            $BODY$;
+
+            create constraint trigger __arches_trg_update_spatial_views
+            after insert or update or delete
+            on spatial_views
+            deferrable initially deferred
+            for each row
+                execute procedure __arches_trg_fnc_update_spatial_views();
+    """
+
+    dependencies = [
+        ("models", "11800_remove_foreign_object"),
+    ]
+
+    operations = [
+        migrations.RunSQL(drop_legacy_trigger, create_legacy_trigger),
+        pgtrigger.migrations.AddTrigger(
+            model_name="spatialview",
+            trigger=pgtrigger.compiler.Trigger(
+                name="__arches_trg_update_spatial_views",
+                sql=pgtrigger.compiler.UpsertTriggerSql(
+                    declare="DECLARE sv_perform text; valid_geom_nodeid boolean; has_att_nodes integer; valid_att_nodeids boolean; valid_language_count integer;",
```

Comment:
For instance: This is the chewed up SQL that gets run, ensuring that 0001 and 0003 run distinct SQL, but the idea is that the developer would work with the single source of truth .sql file

Reply:
Ok - so if I'm understanding correctly, the .py file would be a chewed up copy of what's in the .sql file?

Reply:
And sometimes there wouldn't be a SQL file if the developer never wrote SQL by hand in the first place

Reply:
re: conversation with @jacobtylerwalls to produce the full migration sql, you can run the `sqlmigrate` subcommand

`python manage.py sqlmigrate models [migration_name]`